app/controllers/tutorials_controller.rb

```diff
@@ -1,9 +1,6 @@
 class TutorialsController < ApplicationController
-  def index
-    @tutorials = Tutorial.includes(:tutorial_steps).all
-  end
-
   def show
-    @tutorial = Tutorial.includes(:tutorial_steps).find(params[:id])
+    @tutorials = Tutorial.includes(:tutorial_steps).all
+    @tutorial = params[:id] ? @tutorials.find { |tutorial| tutorial.slug == params[:id] } : @tutorial = @tutorials.first
```

Comment:
In playing around with this locally, I think there's a couple things to investigate…

1. `@tutorials.find` might be triggering another hit to the database. `@tutorials.to_a.find` should alleviate that.
2. The second `@tutorial` assignment might be redundant.

This might do the trick:

```rb
@tutorial = params[:id] ? @tutorials.to_a.find { |tutorial| tutorial.slug == params[:id] } : @tutorials.first
```

Reply:
1. According to this stackoverflow answer, that shouldn't trigger another DB call. How do you tell? https://stackoverflow.com/questions/31296160/rails-find-a-record-within-activerecordrelation-object-without-querying-databa
2. Good catch, that was a typo

Reply:
And... using rails console it appears that the first stackoverflow answer is wrong. But the second using "detect" works.